flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/metrics/KubernetesClientMetrics.java

```diff
@@ -118,32 +139,91 @@ public KubernetesClientMetrics(
                     createMeterViewForMetricsGroup(
                             responseMetricGroup.addGroup(HTTP_RESPONSE_5XX)));
         }
+        this.requestStartTimes = new ConcurrentHashMap<>();
     }
 
     @Override
-    public Response intercept(Chain chain) throws IOException {
-        Request request = chain.request();
-        updateRequestMetrics(request);
-        Response response = null;
-        final long startTime = System.nanoTime();
-        try {
-            response = chain.proceed(request);
-            return response;
-        } finally {
-            updateResponseMetrics(response, startTime);
+    public AsyncBody.Consumer<List<ByteBuffer>> consumer(
```

Comment:
The `consumer` method is not called when building websockets, thus the new implementation outputting different request count than before originally.

I think we could use `before` instead since we don't modify the consumer here anyways. That one will be called for websockets as well.

https://github.com/fabric8io/kubernetes-client/blob/37ee45b94659879718fae7e9b77c2357a9e10c28/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/http/StandardHttpClient.java#L264

Reply:
Argh. The problem with before is that I'm the request.id we see might not be the actual one sent as another filter can modify it after the invocation of our before callback.

I'll have to have another poke around the latency tracking.

Reply:
I've pushed up https://github.com/apache/flink-kubernetes-operator/pull/863/commits/c4c9edaeaa66830a24523e5a996a8c98abc0511a which (ab)uses request headers to tack a start time onto a request and read it back when the request completes. I've also added a  test which triggers websockets to ensure they are counted.